### PR TITLE
BugFix: fixes awaitables and broad-exceptions in garbage-collector

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Another group of fixes from PR #2079 associated with issues in the garbage collector.

- broad exceptions catches were suppressing ``CancelledError`` which avoided the gc. Since the gc is a periodic background task it could never be cancelled
- some extra fixes in garbage_collector.py
- some coroutines were not awaited

## How to test

```cmd
make devenv
cd packages/postgres-database
make install-dev
make tests
cd ...
cd services/web/server
make install-dev
make test-dev-unit
```
cover this bugfix

### CODE REVIEW

**Review by mguidon — APPROVED** [review]

**Review by odeimaiz — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py [hunk 1]
```diff
@@ -42,32 +44,37 @@
...
-    yield
-    task = app[APP_GARBAGE_COLLECTOR_KEY]
-    task.cancel()
-    await task
+def setup_garbage_collector(app: web.Application):
+    async def _setup_background_task(app: web.Application):
+        # on_startup
+        # create a background task to collect garbage periodically
+        loop = asyncio.get_event_loop()
+        cgp_task = loop.create_task(collect_garbage_periodically(app))
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py:61]: Could you please rename `cgp_task` to `periodic_garbage_collect_task` or similar?

File: services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py [hunk 2]
```diff
@@ -42,32 +44,37 @@
...
 from .config import (
     APP_CLIENT_REDIS_LOCK_KEY,
-    APP_GARBAGE_COLLECTOR_KEY,
     GUEST_USER_RC_LOCK_FORMAT,
     get_garbage_collector_interval,
 )
 from .registry import RedisResourceRegistry, get_registry
 
 logger = logging.getLogger(__name__)
+database_errors = (psycopg2.DatabaseError, asyncpg.exceptions.PostgresError)
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py:53]: Could you also consider renaming `database_errors ` to `GroupedDatabaseErrors` ?

**Review by GitHK — APPROVED** [review]
Very nice, I see you reduced the scope of error catching in the GC to only database errors.

I have only 2 rename requests

File: services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py (hunk 2 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py:53]: it is not a class therefore it has to be snake-case.

File: services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py (hunk 1 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py:61]: yes. will do in next PR. I want to get thisone in so I can push py3.8 upgrade

**Pull request merged**